cloud_init.rb

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+
+# Copyright (C) 2015 Oregon State University
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+require 'yaml'
+
+host = ENV['INSTANCE_NAME']
+user = ENV['CINIT_USER']
+ssh_key = ENV['CINIT_SSH_KEY']
+manage_resolv_conf = ENV['CINIT_MANAGE_RESOLV_CONF']
+name_servers = ENV['DNS_SERVERS']
+search_domains = ENV['DNS_SEARCH']
+domain = host.split('.').last(2).join('.')
```

Comment:
Nice catch. I was mainly wanting to automatically set the domain based on the instance FQDN. What would be a better method for doing that to catch situations like that?